test/libsolidity/syntaxTests/types/hex_literal_to_bytesXX_same_size_implicit.sol

```diff
@@ -0,0 +1,13 @@
+contract C {
+   function f() public pure {
+     bytes1 b1 = 0x01;
+     bytes1 b2 = 0xFF;
+     bytes2 b3 = 0x100;
+     bytes2 b4 = 0xFFFF;
+     bytes3 b5 = 0x10000;
+     bytes3 b6 = 0xFFFFFF;
+     bytes4 b7 = 0x1000000;
```

Comment:
I think we should even force the leading zero, if you have a long hex constant, you don't really count if the number of hex digits is even or odd.